src/dynamic_instruction.rs

```diff
@@ -0,0 +1,14 @@
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub enum DynamicInstruction {
+    /// Load native program by name
+    LoadNative { name: String },
+
+    /// Load BPF program by name
+    LoadBpfFile { name: String },
+
+    /// Load BPF program by file bits
+    LoadBpf { offset: u64, prog: Vec<u8> },
+
+    /// Call a program
+    Call { input: Vec<u8> },
```

Comment:
- Distinguishes between the other instructions and helps in error checking/bad data handling
- Makes the operation explicit about what is being performed

Anything specific you were concerned with?

Reply:
I guess I just don't understand why as a user I couldn't "Call" a dynamic program in the same way I "Call" the built-in programs like System, Budget, etc.

Reply:
I'm not following, to call System you must call one of the pre-defined operations (create, assign, move, ...) and for Budget the same (new_signature, new_taxed, ...).  For dynamic programs its once removed since we can't define exactly what the user's program does.  All we know is we can call it so the user would use `DynamicInstruction::Call` to call it it and pass whatever custom instruction they've defined for their program.  The user could even define their own instruction/transaction layer for their program that internally would call `DynamicInstruction::Call`